### DISCUSSION TITLE
Realtime Updates

### DISCUSSION THREAD

**Discussion by itsobi** [supabase/supabase#30904] [Questions]
I'm trying to implement real-time updates in my application. Everything is working up until line 61. Is there anything in the docs I can specifically look at to solve this?

Thanks

![supabase-realtime](https://github.com/user-attachments/assets/0c783017-a9e4-4981-a25a-de0f6ae2bcd1)

**Comment by GaryAustin1** [comment]
You don't say what is wrong or anything about error...

You can use `.subscribe(status=>console.log(status))` to get some info on connection status.
You seem to be using Clerk.   If are minting your own JWT's and setting the Authorization header directly you will need to set the Realtime authorization header.  https://supabase.com/docs/guides/realtime/postgres-changes?queryGroups=language&language=js#custom-tokens

Also be careful with subscriptions in useEffect.  If it gets called on a page transition with the same channel name you will have a conflict with the channels conflicting as opening and closing will be async.
